File: client/src/app/models/board.ts
export type Board = {
    id: number;
    name: string;
    workspaceId: number;
    createdAt: string;
    tasks: TaskItem[];
}

export type TaskItem = {
    id: number;
    title: string;
    description?: string;
    status: string;
    position: number;
    assigneeId?: string;
    assigneeEmail?: string;
    dueDate?: string;
    labels?: string;
    boardId: number;
    createdAt: string;
    updatedAt: string;
}

File: API/Entities/Board.cs
namespace API.Entities;

public class Board
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int WorkspaceId { get; set; }
    public Workspace Workspace { get; set; } = null!;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<TaskItem> Tasks { get; set; } = [];
}
